### PR TITLE
added docker version check

### PR DESCRIPTION
**Pull request by djosborne**

### CODE REVIEW

File: calico_containers/calicoctl.py [hunk 1]
```diff
@@ -467,6 +477,25 @@ def checksystem(fix=False, quit_if_error=False):
...
+        system_ok = False
+    else:
+        api_version = normalize_version(info['ApiVersion'])
+        # Check that API Version is at least 1.16
+        if cmp(api_version, normalize_version("1.16")) < 0:
+            print >> sys.stderr, "ERROR: Docker server must support Docker " \
+                                 "Remote API v1.16 or greater."
+            system_ok = False
+
+        # Check that if Remote API Version is 1.20, they are using the experimental docker build
```
Comment (djosborne) [inline calico_containers/calicoctl.py]: @tomdee Is this the desired behavior?

File: calico_containers/calicoctl.py [hunk 2]
```diff
@@ -467,6 +477,25 @@ def checksystem(fix=False, quit_if_error=False):
             print >> sys.stderr, "WARNING: ipv6 forwarding is not enabled."
             system_ok = False
 
+    # Check docker comparability
+    try:
+        info = docker_client.version()
+    except docker.errors.APIError:
+        print >> sys.stderr, "ERROR: Docker server must support Docker Remote API v1.16 or greater."
```
Comment (tomdee) [inline calico_containers/calicoctl.py]: Use the docker version constant

File: calico_containers/calicoctl.py (hunk 1 above)
Comment (tomdee) [inline calico_containers/calicoctl.py]: No. As discussed remove the 1.20 block when checking into master. Then when we merge master into libnetwork-rebased we'll update this code.

**Comment by tomdee** [comment]
LGTM - a couple of minor comments. Fix those up then merge.

**Pull request merged**